eng/pipelines/doc-index.yml

```diff
@@ -11,40 +11,62 @@ jobs:
       - pwsh: |
           Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
           -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
-          echo "##vso[task.setvariable variable=docfxPath;isOutput=true]$(Build.SourcesDirectory)/docfx/docfx.exe"
-        workingDirectory: $(Build.SourcesDirectory)
+          echo "##vso[task.setvariable variable=docfxPath]$(Build.BinariesDirectory)/docfx/docfx.exe"
+        workingDirectory: $(Build.BinariesDirectory)
         displayName: Download and Extract DocFX
-        name: setupDocfxTool
       - task: PowerShell@2
         displayName: "Generate Doc Index"
         inputs:
           pwsh: true
           filePath: $(Build.SourcesDirectory)/eng/docs/index/Generate-DocIndex.ps1
           arguments: >
-            -Docfx $(setupDocfxTool.docfxPath)
+            -Docfx $(docfxPath)
             -RepoRoot $(Build.SourcesDirectory)
             -DocGenDir "$(Build.SourcesDirectory)/eng/docs/index"
+            -DocOutDir "$(Build.ArtifactStagingDirectory)/docfx_project"
             -verbose
-
       - task: UsePythonVersion@0
         displayName: 'Use Python 3.6'
         inputs:
           versionSpec: '3.6'
 
       - template: /eng/common/pipelines/templates/steps/mashup-doc-index.yml
         parameters:
-          SourceDirectory: $(Build.SourcesDirectory)
+          SourceDirectory: $(Build.ArtifactStagingDirectory)
 
-      - task: CopyFiles@2
-        displayName: Copy HTML to Artifacts Directory
-        inputs:
-          sourceFolder: $(Build.SourcesDirectory)/docfx_project/
-          content: '**\*'
-          targetFolder: $(Build.ArtifactStagingDirectory)/docfx_project
-          overWrite: true
+      - pwsh: |
+          Copy-Item -Path $(Build.SourcesDirectory)/eng/common/scripts/git-branch-push.ps1 -Destination ./ -Recurse -Force
```

Comment:
Lets be sure to copy the entire "eng" directory here as well.

Reply:
Sure. If so, we will have the eng/common folder show up in gh-pages.

Reply:
Or we put the entire folder under `$(Build.BinariesDirectory)`